src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerFactoryExtensions.cs

```diff
@@ -67,26 +67,30 @@ public static ILoggerFactory AddApplicationInsights(this ILoggerFactory factory,
             Func<string, LogLevel, bool> filter,
             Action loggerAddedCallback)
         {
-            var client = serviceProvider.GetService<TelemetryClient>();
-            var debugLoggerControl = serviceProvider.GetService<ApplicationInsightsLoggerEvents>();
-            var options = serviceProvider.GetService<IOptions<ApplicationInsightsLoggerOptions>>();
-
-            if (options == null)
+            var loggerMarker = serviceProvider.GetService<LoggerMarker>();
```

Comment:
ConfigureServices() is guaranteed to be called before Configure() where this method is invoked, ensuring that LoggerMarker will be present by the code this code is executed.